automated/linux/kselftest/kselftest.yaml

```diff
@@ -60,5 +69,5 @@ params:
 run:
     steps:
         - cd ./automated/linux/kselftest/
-        - ./kselftest.sh -c "${TST_CMDFILES}" -T "${TST_CASENAME}" -t "${TESTPROG}" -s "${SKIP_INSTALL}" -u "${TESTPROG_URL}" -L "${SKIPLIST}" -S "${SKIPFILE}" -b "${BOARD}" -g "${BRANCH}" -e "${ENVIRONMENT}" -p "${KSELFTEST_PATH}" -n "${SHARD_NUMBER}" -i "${SHARD_INDEX}"
+        - ./kselftest.sh -c "${TST_CMDFILES}" -T "${TST_CASENAME}" -t "${TESTPROG}" -s "${SKIP_INSTALL}" -u "${TESTPROG_URL}" -L "${SKIPLIST}" -S "${SKIPFILE}" -b "${BOARD}" -g "${BRANCH}" -e "${ENVIRONMENT}" -p "${KSELFTEST_PATH}" -n "${SHARD_NUMBER}" -i "${SHARD_INDEX}" -E "${ENV}"
```

Comment:
So for me to add multiple arbitrary environment variables I have to do:
ENV: "argh=var -E argh2=var2 -E arg3=var3" is that correct?

For constancy, can't we remove '-E "${ENV}"' and just have "${ENV}" here and then the user can specify
ENV: "-E argh=var -E argh2=var2 -E arg3=var3" thoughts?

Reply:
I have tried the way you suggested but due to how bash expands variables it doesn't seem to work if you alter the step to look like `- ./kselftest.sh ${ARGS}`, spaces in `ARGS` dont get preserved and if you do `"${ARGS}" it will all get treated as single arg by the script. 

If the user wants to provide additional `-E` arguments they need to do it to the step command with a different parameter name like:

```
params:
  ENV_1: "Value 1"
  ENV_2: "Value 2"
  ENV_3: "Value 3"

run:
  steps:
    - ./kselftest.sh -E "${ENV_1}" -E "${ENV_2}" -E "${ENV_3}"
```

Alternative solution if we want to support multiple environment variables using a single parameter is:
```
params:
  ENV: ENV_1="Value 1" ENV_2="Value 2" ENV_3="Value 3"
run:
  steps:
    - ./kselftest.sh -E "${ENV_1}"
```

and in the `kselftest.sh` we can directly have the following:
```
        E)
            if [ -n "${OPTARG}" ]; then
                echo "executing 'eval ${OPTARG}'"
                eval "export ${OPTARG}"
            fi
            ;;
```

Reply:
Reading the help text for ENV I was under the impression that you wanted to support multiple ENVs. If now, maybe we should just reword the helptext a bit to support 1 ENV only ?

Reply:
That looks good, thank you @crazoes